tools/haskell-language-server/default.nix

```diff
@@ -13,7 +13,10 @@
 }:
 let
   ghcVersion = project.project.pkg-set.options.compiler.nix-name.value;
-  index-state = project.project.index-state;
+  index-state =
+    if builtins.hasAttr "index-state" project
+    then project.index-state
+    else null;
```

Comment:
Did I overlook an easier builtin for this?  This is silly, but I see the idiom all over.

Reply:
not sure if it's more readable but also 
```nix
if project ? index-state then project.index-state else null
```
i think works